sunpy/coordinates/utils.py

```diff
@@ -80,11 +84,21 @@ class GreatArc:
     """
 
     def __init__(self, start, end, center=None, points=None):
+
+        # Observer
+        self.observer = start.observer
+
+        # Co-ordinate frame of the starting point
+        self.start_frame = start.frame
+
+        # Observation time
+        self.obstime = start.obstime
+
         # Start point of the great arc
-        self.start = start
+        self.start = start.transform_to(Heliocentric)
 
         # End point of the great arc
-        self.end = end
+        self.end = end.transform_to(Heliocentric(observer=self.observer))
```

Comment:
You don't specify the `obstime` for the destination frame.  But instead of fixing that, you should probably write `end.transform_to(self.start.frame)`, which makes it more clear what you are doing.